From 3444ba44d2cd70bb6686f0c0d78fe051f16e979a Mon Sep 17 00:00:00 2001
From: Hood Chatham <roberthoodchatham@gmail.com>
Date: Sun, 3 Sep 2023 09:00:42 -0700
Subject: [PATCH] clang 17 compatibility: filter out __FPCLASS_*

Backport of a change in the main branch of wasi-libc which hasn't been released yet.

---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index f350ecb..674aba4 100644
--- a/Makefile
+++ b/Makefile
@@ -702,6 +702,7 @@ check-symbols: startup_files libc
 	    | sed -e 's/__GNUC_VA_LIST $$/__GNUC_VA_LIST 1/' \
 	    | grep -v '^#define __\(BOOL\|INT_\(LEAST\|FAST\)\(8\|16\|32\|64\)\|INT\|LONG\|LLONG\|SHRT\)_WIDTH__' \
 	    | grep -v '^#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_\(1\|2\|4\|8\)' \
+	    | grep -v '^#define __FPCLASS_' \
 	    > "$(SYSROOT_SHARE)/predefined-macros.txt"
 
 	# Check that the computed metadata matches the expected metadata.
-- 
2.25.1

